lib/libzfs/libzfs_sendrecv.c

```diff
@@ -1007,8 +1007,17 @@ send_print_verbose(FILE *fout, const char *tosnap, const char *fromsnap,
 			(void) fprintf(fout, dgettext(TEXT_DOMAIN,
 			    "incremental\t%s\t%s"), fromsnap, tosnap);
 		} else {
+#if defined(__GNUC__) && !defined(__clang__) && \
+	defined(ZFS_GCC_FORMAT_OVERFLOW_SUPPORTED)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-overflow"
+#endif
 			(void) fprintf(fout, dgettext(TEXT_DOMAIN,
 			    "full\t%s"), tosnap);
+#if defined(__GNUC__) && !defined(__clang__) && \
+	defined(ZFS_GCC_FORMAT_OVERFLOW_SUPPORTED)
```

Comment:
What error is being printed that makes this needed here, but not for other fprintf() functions in this function? This is inconsistent. I understand that it is the compiler being inconsistent here, but the different treatment of these function calls is confusing to me.

I wonder if the issue can be suppressed with some minor alterations to the fprintf() line like what was done in cmd/zfs/zfs_main.c.